numpy/lib/npyio.py

```diff
@@ -87,6 +87,13 @@ def __dir__(self):
 
 
 def zipfile_factory(*args, **kwargs):
+    """Create a zipfile.Zipfile. Allows for Zip64, as well as pathlib.Path 
+    objects."""
+    assert len(args) >= 1, "zipfile_factory requires a file"
```

Comment:
Note that `assert` doesn't work when python is run in optimized mode. Just raise an error if you want to do it that way. Personally, I think a `TypeError` from `ZipFile` is fine, especially as it is unlikely to ever be seen.